pytorch_lightning/plugins/ddp_sequential_plugin.py

```diff
@@ -0,0 +1,411 @@
+            around a potential deadlock in pytorch when using tensor parallelism
+            at the same time. Defaults to `True` if
+            `get_model_parallel_world_size() > 1`
+        """
+        self._check_pipe_available()
+        super().__init__(**kwargs)
+
+        self.balance = balance
+
+        self.microbatches = microbatches
+        self.checkpoint = checkpoint
+        self.balance_mode = balance_mode
+        self.pipelined_backward = pipelined_backward
+        self.main_rpc_process = False  # Updated by main process, default for all secondary processes
+
+    def init_ddp_connection(
+            self,
+            trainer,
+            cluster_environment,
+            global_rank: int,
+            world_size: int,
+            is_slurm_managing_tasks: bool = True,
+    ) -> None:
+        trainer.prepared_for_backwards = False
+        self._check_arguments(trainer)
+        if self._skip_init_connections(trainer):
+            return
+        super().init_ddp_connection(
+            trainer=trainer,
+            cluster_environment=cluster_environment,
+            global_rank=global_rank,
+            world_size=world_size,
+            is_slurm_managing_tasks=is_slurm_managing_tasks
+        )
+        super().init_rpc_connection(
+            global_rank=global_rank,
+            world_size=world_size
+        )
+        model = trainer.get_model()
+        self.gpus_per_model = self._infer_check_num_gpus(trainer)
+        self.init_model_parallel_groups(trainer)
+        self.set_main_rpc_process()
+
+        self._check_sequential_model_exists(model)
+        if self.main_rpc_process:
+            if self.balance is None:
+                self._infer_model_balance(trainer)
+            self._assert_valid_model_balance(trainer)
+
+    def on_before_manual_backward(self, model: LightningDistributedDataParallel, output: Any):
+        pass
+
+    def _infer_model_balance(self, trainer):
+        log.info(f'Inferring model balance using {self.balance_mode} mode')
+        model = trainer.get_model()
+        if model.example_input_array is None:
+            raise MisconfigurationException(
+                'Please set example_input_array to your model, so we can infer the right model balance for you')
+        balance_func = getattr(pipe_balance, self.balance_mode)
+        self.balance = balance_func(self.gpus_per_model, model.sequential_module, model.example_input_array)
+        self._sync_balance_to_all_parallel_groups()
+
+        log.info(f'The following model balance {self.balance.tolist()} was inferred using {self.balance_mode} mode')
+
+    def _sync_balance_to_all_parallel_groups(self, main_rank=0):
+        """
+        Ensures that we sync the balance to all main processes, so that the balance is the same per replica.
+        Args:
+            main_rank: The rank with the balance we'd like to replicate.
+        """
+        self.balance = torch.tensor(self.balance, dtype=torch.int).cuda()
```

Comment:
Can you use `to='cuda'` instead of creating a cpu tensor and then moving it to cuda?